proto/regen/ecocredit/v1alpha1/tx.proto

```diff
@@ -274,4 +287,133 @@ message MsgUpdateClassMetadata {
 }
 
 // MsgUpdateClassMetadataResponse is the MsgUpdateClassMetadata response type.
-message MsgUpdateClassMetadataResponse {}
\ No newline at end of file
+message MsgUpdateClassMetadataResponse {}
+
+// MsgSell is the Msg/Sell request type.
+message MsgSell {
+
+  // owner is the address of the owner of the credits being sold.
+  string owner = 1;
+
+  // orders are the sell orders being created.
+  repeated Order orders = 2;
+
+  // Order is the content of a new sell order.
+  message Order {
+
+    // batch_denom is the credit batch being sold.
+    string batch_denom = 1;
+
+    // quantity is the quantity of credits being sold from this batch. If it is
+    // less then the balance of credits the owner has available, the order will
+    // be adjusted downwards to the owner's balance.
+    string quantity = 2;
+
+    // ask_price is the price the seller is asking for each unit of the
+    // batch_denom. Each credit unit of the batch will be sold for at least the
+    // ask_price or more.
+    cosmos.base.v1beta1.Coin ask_price = 3;
+
+    // dont_auto_retire disables auto-retirement of credits which allows a
+    // buyer to disable auto-retirement in their buy order enabling them to
+    // resell the credits to another buyer.
+    bool dont_auto_retire = 4;
```

Comment:
Can we just call it `auto_retire` and use it otherwise. `dont` prefix is bit odd

Reply:
My understanding is that we generally want to default to auto-retirement. Maybe `disable_auto_retirement` is clearer?

Reply:
Yes, that sounds better